addons/stock_account/models/product.py

```diff
@@ -855,17 +859,32 @@ def _get_stock_account_property_field_names(self):
             'property_stock_valuation_account_id',
         ]
 
+    def _inverse_stock_account_property(self):
```

Comment:
@odoo/rd-logistics 
May be you can help me check the refactor for those property fields

Data generated by code before was not consitent in the database
From the code I read, logically you want the fallback of `property_valuation` to be `'manual_periodic'`
but the fallback of `property_xxx` to be the account when `property_valuation` is changed to `'real_time'`
so when a new `product.category` is created without specification for these field
you will get `product_category.property_valuation == 'manual_periodic'` but `product_category.property_xxx == fallback_account`
which is inconsistent with this idea of `_check_valuation_accounts` where you seems want `product_category.property_xxx` to be empty recordset when `product_category.property_valuation == 'manual_periodic'`

after this PR and upgrade script
there might be many values like `'{"1": NULL}'::jsonb` in the column cause by the `_inverse_stock_account_property` which is logically equvilant to the old `_check_valuation_accounts`.
The inconsistence problem still exists. For example when read with company 2, the `'{"1": NULL}'::jsonb` will fallback to a fallback_account but the `product_category.property_valuation == 'manual_periodic'`
Also since the column value `'{"1": NULL}'::jsonb` IS NOT NULL, searching this field might be slower

My proposal
1. remove the whole `write` part of the `_check_valuation_accounts`, which is `_inverse_stock_account_property` in the new code.
2. check everywhere those property_xxx is used and check `product_category.property_valuation == 'manual_periodic'` before read/search the `property_xxx`
	For example:
		1. `search([('property_stock_account_input_categ_id', '=', 100)])` should become 
			`search([('property_stock_account_input_categ_id', '=', 100), 'property_valuation', '=', manual_periodic)])`
		2. `account = product_category.property_stock_account_input_categ_id` should become
			`account = product_category.property_stock_account_input_categ_id if product_category.property_valuation == 'real_time' or False`
3. use UI to hide `property_xxx` fields when `product_category.property_valuation == 'real_time'`

Or you can make the current `property_xxx` computed but you may need to rename the columns and add support for `search` and `inverse`

Reply:
@HydrionBurst The account fields on `product.category` are relevant only in case of `property_valuation == 'real_type'` as you can see here:

https://github.com/odoo/odoo/blob/a711f4e620a9ea808f7b4dfc22428df610f95d2f/addons/stock_account/views/product_views.xml#L40-L43

Also the default value for `property_valuation` should be `manual_periodic` so the account can be `False` by default but if we switch from `real_time` to `manual_periodic`, we would like to have a value in each account field